src/core/BBoxOffset.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react'
+import { Box3, Vector3, Object3D } from 'three'
+import { useFrame, GroupProps } from '@react-three/fiber'
+
+export interface BBOffsetProps extends GroupProps {
+  anchor: Vector3 | [number, number, number]
+}
+
+export const BBoxOffset = ({ anchor, ...props }: BBOffsetProps) => {
+  const ref = React.useRef<Object3D>(null!)
+  const [parentRef, setParentRef] = React.useState<Object3D | null>(null)
+
+  // Reattach group created by this component to the parent's parent,
+  // so it becomes a sibling of its initial parent.
+  // We do that so the children have no impact on a bounding box of a parent.
+  React.useEffect(() => {
+    if (ref.current?.parent?.parent) {
+      setParentRef(ref.current.parent)
+      ref.current.parent.parent.add(ref.current)
+    }
+  }, [])
+
+  // Do not create objects in loop.
+  // https://docs.pmnd.rs/react-three-fiber/advanced/pitfalls#don't-re-create-objects-in-loops
+  const boundingBox = new Box3()
+  const boundingBoxSize = new Vector3()
```

Comment:
Lets move these 2 variables outside the component or make them refs. That way the objects aren't initialized on every re-render.

Reply:
Sure, my bad